pkg/reconciler/pipelinerun/affinity_assistant.go

```diff
@@ -68,6 +70,37 @@ func (c *Reconciler) createAffinityAssistants(ctx context.Context, wb []v1beta1.
 				if err == nil {
 					logger.Infof("Created StatefulSet %s in namespace %s", affinityAssistantName, namespace)
 				}
+			// check whether the affinity assistant (StatefulSet) exists and the affinity assistant pod is created
+			// this check requires the StatefulSet to have the readyReplicas set to 1 to allow for any delay between the StatefulSet creation
+			// and the necessary pod creation, the delay can be caused by any dependency on PVCs and PVs creation
+			// this case addresses issues specified in https://github.com/tektoncd/pipeline/issues/6586
+			case err == nil && a != nil && a.Status.ReadyReplicas == 1:
+				if nodes == nil {
+					nodes, err = c.KubeClientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{
+						FieldSelector: "spec.unschedulable=true",
+					})
+					if err != nil {
+						errs = append(errs, fmt.Errorf("could not get the list of nodes, err: %w", err))
+					}
+				}
+				for _, n := range nodes.Items {
+					// get the pod created for a given StatefulSet
+					// pod is assigned ordinal of 0 with the replicas set to 1
+					p, err := c.KubeClientSet.CoreV1().Pods(pr.Namespace).Get(ctx, a.Name+"-0", metav1.GetOptions{})
+					// ignore instead of failing if the affinity assistant pod was not found
+					if err != nil && !apierrors.IsNotFound(err) {
+						errs = append(errs, fmt.Errorf("could not get the affinity assistant pod for StatefulSet %s: %w", a.Name, err))
```

Comment:
It looks like if there's an error getting the pod we still attempt to delete it, should we just break here instead?

Reply:
please let me know if you see the same issue after I have updated the implementation to address https://github.com/tektoncd/pipeline/pull/6596#discussion_r1194328882

The idea here is to error out if `Get` returns an error but that error is not a `pod not found error`. Like we discussed, there could be a transient error in retrieving affinity assistant pod. Ignore such error but any other error from `Get` command results in an error.